lib/spack/spack/directives.py

```diff
@@ -663,39 +663,36 @@ def _execute_patch(pkg_or_dep):
 
 @directive("variants")
 def variant(
-    name,
-    default=None,
-    description="",
-    values=None,
-    multi=None,
-    validator=None,
-    when=None,
-    sticky=False,
+    name: str,
+    *,
+    default: Optional[Any] = None,
+    description: str,
```

Comment:
I get the idea, but would recommend not to make unnecessary breaking changes. If you really care about descriptions, we can add it to audits. But like this it's just more painful to upgrade spack with custom repos.

Reply:
I'm not opposed to changing it to an audit. Let me dig around and see how hard that would be.